PSEUDO API 1.md

```diff
@@ -0,0 +1,166 @@
+
+## XR
+
+	interface XR {
+		Promise<sequence<XRDisplay>> getDisplays();
+	};
+
+
+## XRDisplay
+
+	interface XRDisplay {
+		readonly attribute DOMString displayName;
+		readonly attribute boolean isExternal;
+
+		Promise<boolean> supportsSession(XRSessionCreateParametersInit parameters);
+		Promise<XRSession> requestSession(XRSessionCreateParametersInit parameters);
+
+		attribute EventHandler ondeactivate;
+	};
+
+### Todo
+
+- calibration
+
+## Reality
+
+	interface Reality {
+		readonly atttribute DOMString realityName;
+		readonly attribute boolean hasPointCloud;
+		readonly attribute XRCoordinates stageLocation;
+
+		Promise<boolean> requestStageLocation(float x, float y float z, XRCoordinates)
+		Promise<boolean> requestResetStageLocation()
+
+		Promise<XRPointCloud> getPointCloud()
+
+		Promise<XRLightEstimate> getLightEstimate()
+
+		attribute EventHandler onchange;
+	};
+
+### Todo
+
+- configuration
+- anchors
+- manifold instead of point cloud
+
+## XRPointCloud
+
+	interface XRPointCloud {
+		readonly attribute Float32Array points;
+	}
+
+## XRLightEstimate
+
+	interface XRLightEstimate {
+		readonly attribute float ambientIntensity;
+		readonly attribute float ambientColorTemperature;
+	}
+
+## XRSession
+
+	interface XRSession : EventTarget {
+		readonly attribute XRDisplay display;
+		readonly attribute XRSessionCreateParameters createParameters;
+
+		attribute double depthNear;
+		attribute double depthFar;
+
+		attribute XRLayer layer;
+
+		Promise<sequence <Reality>> getRealities();
+
+		Promise<XRFrameOfReference> requestFrameOfReference(XRFrameOfReferenceType type);
+
+		long requestFrame(XRFrameRequestCallback callback);
+		void cancelFrame(long handle);
+
+		Promise<void> endSession();
+
+		attribute EventHandler onblur;
+		attribute EventHandler onfocus;
+		attribute EventHandler onresetpose;
+		attribute EventHandler onended;
+	};
+
+## XRPresentationFrame
+
+	interface XRPresentationFrame {
+		readonly attribute FrozenArray<XRView> views;
+
+		XRDisplayPose? getDisplayPose(XRCoordinateSystem coordinateSystem);
+	};
+
+## XRCoordinateSystem
+
+	interface XRCoordinateSystem: EventTarget {
+		readonly attribute Coordinates? coordinates; // https://dev.w3.org/geo/api/spec-source.html#coordinates
```

Comment:
No, probably not.  We used an extension of Cesium entities in argon.js.  Cesium entities basically represent things in ECEF (meters relative to center of earth);  in argon.js, we extended this to allow entities to be relative to other entities, so that we could have local coordinate frames that aligned with the local "view" (East Up South).  

The big question is one of "recentering".  When an app runs for a long time, we can't keep using an EUS relative to some point far away, since you local frame changes.  Like ARKit and Hololens, we assert that you should ask for anchors relative to the local frame each time.

The trick is, we need a dynamic local origin to convert to.  There are different ways this can be done;  I'd like to know what Windows Holographic does for long running apps.